### PR TITLE
feat: Safe Apps permissions

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves

Resolves https://github.com/safe-global/web-core/issues/589

## How this PR fixes it
This PR includes the permission system for Safe Apps:

- Adds permissions management support for the safe apps sdk [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255) implementation released on [7.6.0](https://www.npmjs.com/package/@gnosis.pm/safe-apps-sdk/v/7.6.0)
- Adds permission support for allowing [Browser features](https://www.permissionspolicy.com/) in Safe Apps 

[Designs](https://www.figma.com/file/RqEZSC19sJRH6rbOFm6kxj/Assistance-pop-ups?node-id=222%3A2298)
[RFC](https://docs.google.com/document/d/1hHjoHnwjhq3xtyBuIogFkglS9f6GLsbZTbqZsxduVNE)

## How to test it
Check the tests included here https://github.com/safe-global/safe-react/pull/4019

### CODE REVIEW

File: src/components/safe-apps/AppFrame/useAppCommunicator.ts [hunk 1]
```diff
@@ -32,22 +36,26 @@ type JsonRpcResponse = {
   error?: string
 }
 
-type UseAppCommunicatorConfig = {
-  app?: SafeAppData
+type UseAppCommunicatorHandlers = {
```
Comment (yagopv) [inline src/components/safe-apps/AppFrame/useAppCommunicator.ts:39]: Changing this a little bit thinking in a future separation for this hook to a separate package. Now mostly of the dependences are coming from the SDK so should be easier

**Comment by yagopv** [comment]
Please @JagoFigueroa hit hard this using the test definitions in https://github.com/safe-global/safe-react/pull/4019

**Review by katspaugh — APPROVED** [review]

**Comment by katspaugh** [comment]
With Typography, Box and some other components, you can do this:
```
<Typography
      sx={{
         mx: 8,
       }}
```
->
```
<Typography mx={8}
```

**Comment by yagopv** [comment]
> With Typography, Box and some other components, you can do this:

Oh great, didn't know is possible with Typography 🙌🏻

**Comment by JagoFigueroa** [comment]
All good señores, thanks! ;)

**Pull request merged**